crates/runtime/src/starknet/context.rs

```diff
@@ -127,32 +83,106 @@ pub fn build_transaction_context() -> AccountTransactionContext {
 }
 
 #[must_use]
-pub fn build_context(block_info: BlockInfo) -> EntryPointExecutionContext {
-    let block_context = build_block_context(block_info);
-    let account_context = build_transaction_context();
-
-    EntryPointExecutionContext::new(
-        &block_context,
-        &account_context,
-        ExecutionMode::Execute,
-        false,
-    )
-    .unwrap()
+pub fn build_transaction_context(block_info: &BlockInfo) -> TransactionContext {
+    TransactionContext {
+        block_context: build_block_context(block_info),
+        tx_info: build_tx_info(),
+    }
+}
+
+#[must_use]
+pub fn build_context(block_info: &BlockInfo) -> EntryPointExecutionContext {
+    let transaction_context = Arc::new(build_transaction_context(block_info));
+
+    EntryPointExecutionContext::new(transaction_context, ExecutionMode::Execute, false).unwrap()
 }
 
-#[derive(Copy, Clone, Serialize, Deserialize, Debug)]
-pub struct BlockInfo {
+// Structs copied for serialization
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct ForgeBlockInfo {
     pub block_number: BlockNumber,
-    pub timestamp: BlockTimestamp,
+    pub block_timestamp: BlockTimestamp,
     pub sequencer_address: ContractAddress,
+    pub gas_prices: ForgeGasPrices,
+    pub use_kzg_da: bool,
```

Comment:
X D

Reply:
da is Data Availability if that helps :D